### PR TITLE
feat(event) abstract event watcher, event log

### PR DESCRIPTION
**Pull request by tkmct**
because events and event watchers are heavily depending on platforms,
we need well-abstracted event watcher to implement platform-specific
event watchers.

After fixing Event logs, add Event types and Event watchers.

### CODE REVIEW

File: src/events/types/EventLog.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default interface EventLog {
+  name: string
+  values: any
```
Comment (syuhei176) [inline src/events/types/EventLog.ts]: Is `values` the array of data which is encoded by L1 dependent way? Or structure is different by each L1.

Comment (tkmct) [reply src/events/types/EventLog.ts]: that's a good point!!
this should not be L1 dependent way. We need to implement (Eth|Tez)EventWatcher to return EventLog in the same manner.
Eth can just return `LogDescription` but TezEventWatcher needs to construct values array in poll method.

Comment (tkmct) [reply src/events/types/EventLog.ts]: Then we can reuse other stuff in both implementation.

Comment (syuhei176) [reply src/events/types/EventLog.ts]: EthEventWatcher should know specific event types such as BlockSubmitted event?
I guess we should modify event like this in EthEventWatcher , is it right?

```
const decoded = abi.decode(["uint256", "bytes32"], log.data)
EventLog {
values: [abi.encode(decoded[0]), abi.encode(decoded[0])]
}
```

Comment (tkmct) [reply src/events/types/EventLog.ts]: no, EthEventWatcher doesn't have to know specific EventTypes. Rather, CommitmentContractWatcher has to know the types.
And decoding is responsibility of EventTypes such as `BlockSubmitted`.

Comment (syuhei176) [reply src/events/types/EventLog.ts]: Ok, Cool.

**Pull request merged**